solr/core/src/java/org/apache/solr/request/json/RequestUtil.java

```diff
@@ -214,7 +220,7 @@ public static void processParams(SolrRequestHandler handler, SolrQueryRequest re
           out = "rows";
         } else if (SORT.equals(key)) {
           out = SORT;
-        } else if ("params".equals(key) || "facet".equals(key) ) {
+        } else if ("params".equals(key) || "facet".equals(key) || pluginNames.contains(key) ) {
```

Comment:
It would be simpler to just explicitly allow "funnel" here, but then we'd have to do a similar fork each time we add a new SearchComponent